Source/Noesis.Javascript/JavascriptInterop.cpp

```diff
@@ -387,6 +412,28 @@ JavascriptInterop::ConvertFromSystemArray(System::Array^ iArray)
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////
 
+Handle<Value>
+JavascriptInterop::ConvertFromSystemRegex(System::Text::RegularExpressions::Regex^ iRegex)
+{
+    using RegexOptions = System::Text::RegularExpressions::RegexOptions;
+    if (!iRegex->Options.HasFlag(RegexOptions::ECMAScript))
+        throw gcnew System::Exception("Only regular expressions with the ECMAScript option can be converted.");

```

Comment:
What do you think about throwing an exception here? What are the alternatives?

Reply:
I think this makes sense because it forces users of the library to understand the difference between JS and C# regexps.